src/globus_sdk/_payload.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import abc
+import sys
+import typing as t
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+# TODO: Remove this dispatch after we drop Python 3.8 support.
+#       In 3.9+ `dict.__class_getitem__` is available.
+if t.TYPE_CHECKING:
+    # pylint: disable=unsubscriptable-object
+    _PayloadBaseDict = t.Dict[str, t.Any]
+else:
+    _PayloadBaseDict = dict
+
+
+class Payload(_PayloadBaseDict):
```

Comment:
If we're going to rename `PayloadWrapper`, I think we should push it to be more verbose.

Right now the request structure is such that a client's request method:
* returns a `GlobusHTTPResponse`
* raises a `GlobusAPIError`
* but accepts a `Payload` 

Can we call it `GlobusPayload` or `GlobusHTTPPayload` or `GlobusRequestPayload` or something that qualifies it from other usages of the term?

Reply:
Let's go with `GlobusPayload`. I feel more strongly _against_ `PayloadWrapper` than I do in favor of `Payload`.
My goal isn't to reduce verbosity, but to fix the mismatch of "it's not a wrapper tho!"